### PR TITLE
feat: [datalineage] new module for datalineage

### PR DESCRIPTION
**Pull request by alicejli**
Rerunning script for https://github.com/googleapis/google-cloud-java/pull/8994 as I happened to run it right before the SNAPSHOT release PR went out. 

Command to generate:

> python3.9 generation/new_client/new-client.py generate   --api_shortname=datalineage   --proto-path=google/cloud/datacatalog/lineage   --name-pretty="Data Lineage"   --product-docs="https://cloud.google.com/data-catalog/docs/data-lineage/"   --api-description="Lineage is used to track data flows between assets over time."

### CODE REVIEW

**Comment by meltsufin** [comment]
Can you share the command that was used to generate the library?

File: java-datalineage/.OwlBot.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+deep-copy-regex:
+- source: "/google/cloud/datacatalog/lineage/(v.*)/.*-java/proto-google-.*/src"
+  dest: "/owl-bot-staging/java-datalineage/$1/proto-google-cloud-datalineage-$1/src"
+- source: "/google/cloud/datacatalog/lineage/(v.*)/.*-java/grpc-google-.*/src"
+  dest: "/owl-bot-staging/java-datalineage/$1/grpc-google-cloud-datalineage-$1/src"
+- source: "/google/cloud/datacatalog/lineage/(v.*)/.*-java/gapic-google-.*/src"
+  dest: "/owl-bot-staging/java-datalineage/$1/google-cloud-datalineage/src"
+
+
+api-name: java-datalineage
```
Comment (suztomo) [inline java-datalineage/.OwlBot.yaml:33]: This shouldn't have "java-" prefix. Where did it come from?

https://github.com/googleapis/google-cloud-java/blob/main/java-aiplatform/.OwlBot.yaml#L30

**Review by suztomo — CHANGES_REQUESTED** [review]

**Comment by alicejli** [comment]
> @alicejli api-name having "java-" prefix is strange. What did you run? (It's possible you specified it incorrectly or the script has a bug)
> 
> I'm adding a step to record the Python command: https://github.com/googleapis/google-cloud-java/pull/8995/files#diff-a3adec6fceb9be1a8aca36549237d47cf7b51a8d414ce788a0cb6a8a5aeb7311R192

I ran:

```
python3.9 generation/new_client/new-client.py generate   --api_shortname=datalineage   --proto-path=google/cloud/datacatalog/lineage   --name-pretty="Data Lineage"   --product-docs="https://cloud.google.com/data-catalog/docs/data-lineage/"   --api-description="Lineage is used to track data flows between assets over time."
```

File: java-datalineage/.OwlBot.yaml (hunk 1 above)
Comment (alicejli) [reply java-datalineage/.OwlBot.yaml:33]: I believe it's coming from: https://github.com/googleapis/google-cloud-java/blob/main/generation/new_client/new-client.py#L239 and https://github.com/googleapis/google-cloud-java/blob/main/generation/new_client/templates/owlbot.yaml.monorepo.j2#L33

What is the api-name in this .OwlBot.yaml file used for?

Comment (suztomo) [reply java-datalineage/.OwlBot.yaml:33]: go/owlbot-howto

> api-name: Gets inserted into pull request titles. Useful for mono-repos.

**Comment by suztomo** [comment]
Creating a fix for api-name field in OwlBot.yaml template.

**Comment by suztomo** [comment]
Screenshot 2023-01-19 at 9 27 08 AM

**Comment by alicejli** [comment]
> Screenshot 2023-01-19 at 9 27 08 AM

Well glad to see my newly added check is working :sweat_smile: . 
I think I ran the original script before https://github.com/googleapis/google-cloud-java/pull/8790 was merged in. Let me confirm.

**Comment by alicejli** [comment]
Okay, confirmed that running the script from main does generate the missing lines; I just originally created this before the template was updated. I manually added in the missing lines for this PR.

File: java-datalineage/.OwlBot.yaml (hunk 1 above)
Comment (alicejli) [reply java-datalineage/.OwlBot.yaml]: Thanks. I've manually updated for this PR.

**Review by suztomo — APPROVED** [review]

**Pull request merged**